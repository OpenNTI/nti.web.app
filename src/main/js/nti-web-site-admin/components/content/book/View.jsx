import {Router, Route} from '@nti/web-routing';

import UserBookEnrollment from '../../user-book-enrollment';

import Overview from './overview';
import Reports from './reports';
import Roster from './roster';
import Frame from './Frame';

export default Router.for([
	Route({
		path: '/roster/:userID',
		component: UserBookEnrollment,
		props: {courseContext: true},
		frameless: true,
		getRouteFor (obj, context) {
			if (obj.MimeType === 'application/vnd.nextthought.userbundlerecord' && context === 'site-admin.courses.book-roster.list') {
				return `/roster/${obj.User.getID()}`;
			}

			return null;
		}
	}),
	Route({
		path: '/',
		component: Router.for([
			Route({ path: '/reports', component: Reports, name: 'site-admin.courses.book-reprots' }),
			Route({ path: '/roster', component: Roster, name: 'site-admin.courses.book-roster' }),
			Route({path: '/', component: Overview, name: 'site-admin.courses.book-overview'})
		], {frame: Frame})
	})
]);
