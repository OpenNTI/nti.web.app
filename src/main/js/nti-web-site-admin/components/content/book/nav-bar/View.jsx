import React from 'react';

import Card from '../../../common/Card';

import Identity from './Identity';
import Tabs from './Tabs';

export default function SiteAdminBookNavBar (props) {
	return (
		<Card className="site-admin-course-nav-bar">
			<Identity {...props} />
			<Tabs {...props} />
		</Card>
	);
}
