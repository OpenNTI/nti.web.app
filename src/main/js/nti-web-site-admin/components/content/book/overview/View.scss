@import '~nti-style-common';

.site-admin-book-overview {
	@include clearfix;

	.site-admin-active-days {
		width: 100%;
		margin-top: 5px;
	}

	.active-users {
		width: 100%;

		.dashboard-list-widget {
			margin: 0 0 10px;
		}
	}

	.active-times-widget {
		margin: 0 0 10px;
	}

	.site-admin-active-days {
		margin: 10px 0;
	}

	.site-admin-row {
		display: flex;

		.site-admin-card {
			width: 250px;
			height: 320px;

			.value {
				padding-top: 1rem;
			}
		}

		.active-users {
			height: 320px;

			.header {
				margin-top: 0;
				margin-bottom: 0;
			}
		}
	}
}
