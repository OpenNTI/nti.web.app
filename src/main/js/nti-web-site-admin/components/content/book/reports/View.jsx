import React from 'react';
import PropTypes from 'prop-types';
import { List } from '@nti/web-reports';

import Card from '../../../common/Card';


SiteAdminBooksReports.propTypes = {
	course: PropTypes.object
};
export default function SiteAdminBooksReports ({ course }) {
	return course ?
		(<Card><List context={course} /></Card>) :
		null;
}
