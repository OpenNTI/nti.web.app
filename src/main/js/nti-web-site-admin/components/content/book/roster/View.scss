@import '~@nti/style-common';

.site-admin-course-roster {
	.loading-mask {
		position: relative;
		min-height: 200px;
	}

	ul {
		li {
			box-shadow: 0 1px 0 0 var(--border-grey-light);

			a {
				text-decoration: none;
				display: block;
			}
		}
	}

	.controls {
		span {
			color: var(--primary-blue);
			margin-right: 10px;
			cursor: pointer;
		}
	}
}
