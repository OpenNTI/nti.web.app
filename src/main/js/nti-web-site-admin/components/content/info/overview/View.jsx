import PropTypes from 'prop-types';

import { AdminTools } from '@nti/web-course';

SiteAdminCourseOverview.propTypes = {
	course: PropTypes.object,
};
export default function SiteAdminCourseOverview({ course }) {
	return <AdminTools.Dashboard course={course} />;
}
