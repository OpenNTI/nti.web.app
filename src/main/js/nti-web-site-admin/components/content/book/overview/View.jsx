import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Widgets } from '@nti/web-reports';

const { ActiveDays, ActiveUsers, ActiveTimes } = Widgets;

export default class SiteAdminBookOverview extends React.Component {
	static propTypes = {
		course: PropTypes.object
	}

	state = {}

	componentDidMount () {
		const {course} = this.props;

		if(!this.state.resp && course && course.hasLink('users')) {
			course.fetchLink('users').then(resp => {
				this.setState({resp});
			});
		}
	}

	render () {
		const {course} = this.props;

		return (
			<div className="site-admin-book-overview">
				<div className="site-admin-row">
					<div className="active-users">
						<ActiveUsers entity={course}/>
					</div>
				</div>
				<div className="active-days">
					<ActiveDays entity={course}/>
				</div>
				<div className="active-times">
					<ActiveTimes course={course}/>
				</div>
			</div>
		);
	}
}
