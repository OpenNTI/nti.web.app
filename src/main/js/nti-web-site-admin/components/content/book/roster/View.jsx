import './View.scss';
import React from 'react';
import PropTypes from 'prop-types';
import {List, Loading} from '@nti/web-commons';
import {decorate} from '@nti/lib-commons';
import {scoped} from '@nti/lib-locale';
import {LinkTo} from '@nti/web-routing';
import {searchable} from '@nti/web-search';

import ErrorMessage from '../../../common/ErrorMessage';
import Pager from '../../../common/Pager';
import Card from '../../../common/Card';

import Item from './Item';
import BookRosterStore from './BookRosterStore';

const DEFAULT_TEXT = {
	empty: 'No users have access to this book.',
	loadMore: 'Load More'
};
const t = scoped('nti-site-admin.courses.course.info.Roster', DEFAULT_TEXT);

class SiteAdminBookRoster extends React.Component {
	static propTypes = {
		store: PropTypes.object.isRequired,
		course: PropTypes.object,
		items: PropTypes.array,
		loading: PropTypes.bool,
		error: PropTypes.string,
		numPages: PropTypes.number,
		pageNumber: PropTypes.number
	}

	componentDidMount () {
		this.props.store.loadBook(this.props.course);
	}

	render () {
		const {loading, error, items, store, numPages, pageNumber} = this.props;

		return (
			<div className="site-admin-course-roster">
				{loading && (<div className="loading-mask"><Loading.Mask /></div>)}
				{error && (<ErrorMessage>{t('error')}</ErrorMessage>)}
				{!loading && items && this.renderItems(items)}
				{!loading && !error && <Pager store={store} numPages={numPages} pageNumber={pageNumber}/>}
			</div>
		);
	}

	renderItems (items) {
		if(items.length === 0) {
			return <div className="empty-state">{t('empty')}</div>;
		}

		return (
			<Card>
				<List.Unadorned>
					{items.map((item, index) => {
						return (
							<li key={index}>
								<LinkTo.Object object={item} context="site-admin.courses.book-roster.list">
									<Item item={item} />
								</LinkTo.Object>
							</li>
						);
					})}
				</List.Unadorned>
			</Card>
		);
	}

}

export default decorate(SiteAdminBookRoster, [
	searchable(),
	BookRosterStore.connect({
		loading: 'loading',
		items: 'items',
		error: 'error',
		numPages: 'numPages',
		pageNumber: 'pageNumber'
	})
]);
