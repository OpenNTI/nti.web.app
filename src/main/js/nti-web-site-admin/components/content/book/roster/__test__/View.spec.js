import React from 'react';
import renderer from 'react-test-renderer';

import * as TestUtils from '@nti/web-client/test-utils';
import { flushPromises } from '@nti/lib-commons/test-utils';

import View from '../View';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = numberOfItems => {
	let Items = [];

	for (let i = 0; i < numberOfItems; i++) {
		const index = i + 1;

		Items.push({
			getID: () => 'abc' + index,
			User: {
				Username: 'abc' + index,
				avatarURL: 'someAvatarURL',
			},
		});
	}

	return {
		getBatch: () => {
			return Promise.resolve({
				Items,
				Total: Items.length,
				BatchPage: 1,
			});
		},
	};
};

const onBefore = numberOfItems => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfItems));
};

const onAfter = () => {
	tearDownTestClient();
};

/* eslint-env jest */
describe('Site admin user book roster test (25 items)', () => {
	beforeEach(() => onBefore(25));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const course = {
			getLink: () => 'testLink',
		};

		const cmp = renderer.create(<View course={course} />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin user book roster test (5 items)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const course = {
			getLink: () => 'testLink',
		};

		const cmp = renderer.create(<View course={course} />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin user book roster test (0 items)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const course = {
			getLink: () => 'testLink',
		};

		const cmp = renderer.create(<View course={course} />);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
