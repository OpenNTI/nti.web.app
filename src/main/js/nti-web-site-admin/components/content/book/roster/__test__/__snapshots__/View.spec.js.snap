// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user book roster test (0 items) Basic render test 1`] = `
<div
  className="site-admin-course-roster"
>
  <div
    className="empty-state"
  >
    No users have access to this book.
  </div>
</div>
`;

exports[`Site admin user book roster test (5 items) Basic render test 1`] = `
<div
  className="site-admin-course-roster"
>
  <div
    className="site-admin-card"
  >
    <ul
      className="unadorned-list"
    >
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-7 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-8 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-9 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-10 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-11 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Site admin user book roster test (25 items) Basic render test 1`] = `
<div
  className="site-admin-course-roster"
>
  <div
    className="site-admin-card"
  >
    <ul
      className="unadorned-list"
    >
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-7 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-8 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-9 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-10 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-11 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-0 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-1 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-2 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-3 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-1 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-2 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-3 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-4 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-5 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-6 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-7 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-8 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-9 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-10 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-8 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-9 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-10 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-11 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-0 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
      <li>
        <div
          className="pathLink nti-link-to-path"
          to="#"
        >
          <div
            className="site-admin-course-roster-item"
          >
            <img
              className="avatar avatar-color-1 user-avatar"
              src="someAvatarURL"
            />
            <div
              className="info"
            >
              <span
                className="username username"
                rel="author"
              />
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="site-admin-table-pager"
  >
    <div
      className="page selected"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="page"
      onClick={[Function]}
    >
      2
    </div>
  </div>
</div>
`;
